runtime/compare.c

```diff
@@ -140,9 +140,9 @@ static intnat do_compare_val(struct compare_stack* stk,
     poll_timer = COMPARE_POLL_PERIOD;
     while (--poll_timer > 0) {
       if (v1 == v2 && total) goto next_item;
-      if (Is_long(v1)) {
+      if (Is_long(v1) || v1 == Val_null) {
         if (v1 == v2) goto next_item;
-        if (Is_long(v2))
+        if (Is_long(v2) || v2 == Val_null)
           return Long_val(v1) - Long_val(v2);
```

Comment:
This looks wrong - it'll make Val_null compare equal to Val_int(0).

Reply:
Right, fixed.